contracts/AelinPool.sol

```diff
@@ -385,6 +406,57 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         emit PurchasePoolToken(msg.sender, purchaseTokenAmount);
     }
 
+    /**
+     * @dev allows anyone to become a purchaser with a qualified erc721
+     * nft in the pool depending on the scenarios
+     *
+     * Scenarios:
+     * 1. each wallet holding a qualified NFT to deposit an unlimited amount of purchase tokens
```

Comment:
also we want to support cryptopunks which requires custom if logic embedded in the 721 methods

Reply:
yea sure. will change it and add cryptopunks as well.

Reply:
done. check 96e358261f8e04f73bb908935ba0db1146f14740.